compiler/modules/bir/verify.bal

```diff
@@ -67,18 +67,59 @@ class VerifyContext {
 
 public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode code) returns Error? {
     VerifyContext cx = new(mod, defn);
+    boolean[] tmpRegisterInit = [];
+    if code.registers.length() > 0 {
+        tmpRegisterInit[code.registers.length() - 1] = false;
+    }
     foreach BasicBlock b in code.blocks {
-        check verifyBasicBlock(cx, b);
+        check verifyBasicBlock(cx, b, tmpRegisterInit);
     }
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
-function verifyBasicBlock(VerifyContext vc, BasicBlock bb) returns Error? {
+function verifyBasicBlock(VerifyContext vc, BasicBlock bb, boolean[] tmpRegisterInit) returns Error? {
     foreach Insn insn in bb.insns {
         check verifyInsn(vc, insn);
+        check verifyInsnRegisterKinds(vc, insn, tmpRegisterInit);
+    }
+}
+
+function verifyInsnRegisterKinds(VerifyContext vc, Insn insn, boolean[] tmpRegisterInit) returns Error? {
+    if insn is ResultInsnBase {
+        if tmpRegisterInit[insn.result.number] {
+            return vc.invalidErr("multiple assignments to a tmp register", insn.pos);
+        }
+        else {
+            tmpRegisterInit[insn.result.number] = true;
+        }
+    }
+
+    match insn {
+        var { operand } => {
+            check verifyRegisterKind(vc, operand, tmpRegisterInit, insn.pos);
+        }
+        var { operands } | var { args: operands } => {
+            // JBUG #35557 can't iterate operands
+            Operand[] ops = operands;
+            foreach Operand op in ops {
+                check verifyRegisterKind(vc, op, tmpRegisterInit, insn.pos);
+            }
+        }
+    }
+}
+
+function verifyRegisterKind(VerifyContext vc, Operand r, boolean[] tmpRegisterInit, Position pos) returns Error? {
```

Comment:
Current name doesn't explain what it does. How about `verifyTmpOprandInitialized`